src/conn/conn_api.c

```diff
@@ -1822,9 +1821,40 @@ __conn_single(WT_SESSION_IMPL *session, const char *cfg[])
               WT_SINGLETHREAD_STRING));
     }
 
+    /*
+     * We own the database home, figure out if we're creating it. There are a few files created when
+     * initializing the database home and we could crash in-between any of them, so there's no
+     * simple test. The last thing we do during initialization is rename a turtle file into place,
+     * and there's never a database home after that point without a turtle file. If the turtle file
+     * doesn't exist, it's a create.
+     */
+    WT_ERR(__wt_turtle_exists(session, &exist));
+    conn->is_new = exist ? 0 : 1;
+
+    /*
+     * If the turtle file exists, the WiredTiger file should exist as well, unless we are salvaging.
+     */
+    WT_ERR(__wt_config_gets(session, cfg, "salvage", &cval));
+    if (is_create && !cval.val && !conn->is_new) {
+        WT_ERR(__wt_fs_exist(session, WT_WIREDTIGER, &exist));
+        if (!exist) {
+            F_SET(conn, WT_CONN_DATA_CORRUPTION);
+            WT_ERR_MSG(session, WT_TRY_SALVAGE, "WiredTiger version file cannot be found");
+        } else {
+            WT_ERR(__wt_open(session, WT_WIREDTIGER, WT_FS_OPEN_FILE_TYPE_REGULAR, 0, &fh));
+            WT_ERR(__wt_filesize(session, fh, &size));
+            if ((size_t)size == 0)
+                /*
+                 * If WiredTiger file exists but is size zero, write a message but don't fail.
+                 */
+                WT_ERR(__wt_msg(session, "WiredTiger version file is empty"));
+            WT_TRET(__wt_close(session, &fh));
+        }
+    }
+
     /* We own the lock file, optionally create the WiredTiger file. */
-    ret = __wt_open(
-      session, WT_WIREDTIGER, WT_FS_OPEN_FILE_TYPE_REGULAR, is_create ? WT_FS_OPEN_CREATE : 0, &fh);
+    ret = __wt_open(session, WT_WIREDTIGER, WT_FS_OPEN_FILE_TYPE_REGULAR,
+      is_create || cval.val ? WT_FS_OPEN_CREATE : 0, &fh);
```

Comment:
In the case of performing connection open with salvage, I roughly remember that the file would still be created in this case without the need to check if for salvage. Correct if I'm wrong @fionaochee .

Reply:
The file would only be created if the "create" config is on, so I wanted to handle the case where "salvage" is on but "create" is not.

Reply:
is it possible for salvage to be turned on without create?

Reply:
Yes, I don't think salvage relies on create.`test_config10.py` currently reopens the connection using only salvage.